### PR TITLE
feat: Add config-template support

### PR DESCRIPTION
**Pull request by MinDBreaK**
## New Behavior

The config_templates are supported

## Proposed Release Note Entry

Add support for the config_templates

## Double Check

* [x] I have read the comments and followed the [CONTRIBUTING.md](https://github.com/netbox-community/ansible_modules/blob/devel/CONTRIBUTING.md).
* [x] I have explained my PR according to the information in the comments or in a linked issue.
* [x] My PR targets the `devel` branch.

Notes : I have not yet added tests, I would like to be reviewed before changing anything, and made sure that it would be accepted.

### CODE REVIEW

File: plugins/modules/netbox_config_template.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  - pynetbox
+version_added: "3.15.0"
+extends_documentation_fragment:
+  - netbox.netbox.common
+options:
+  data:
+    required: true
+    type: dict
+    description:
+      - Defines the route target configuration
```
Comment (rodvand) [inline plugins/modules/netbox_config_template.py:30]: Make sure you update the descriptions :)

File: plugins/modules/netbox_config_template.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        data:
+          name: "thisIsMyTemplateName"
+          tags:
+            - Cloud
+          template_code: |
+            #cloud-config
+            packages: 
+              - ansible
+
+    - name: Delete Route Targets
```
Comment (rodvand) [inline plugins/modules/netbox_config_template.py:79]: Just delete the examples not valid for your code.

**Comment by rodvand** [comment]
@MinDBreaK  This looks great - if you can just add a test or two it should be fine to merge.

File: plugins/modules/netbox_config_template.py (hunk 2 above)
Comment (MinDBreaK) [reply plugins/modules/netbox_config_template.py:79]: Ah I thought I removed them, maybe I did not included them in the commit, I'll fix it, thanks !

**Pull request merged**